msdk-io/msdk-io-mzml2/src/main/java/io/github/msdk/io/mzml2/util/MzMLPeaksDecoder.java

```diff
@@ -80,48 +78,52 @@
 
     // first check for zlib compression, inflation must be done before
     // NumPress
-    switch (binaryDataInfo.getCompressionType()) {
-      case ZLIB:
-      case NUMPRESS_LINPRED_ZLIB:
-      case NUMPRESS_POSINT_ZLIB:
-      case NUMPRESS_SHLOGF_ZLIB:
-        iis = new InflaterInputStream(is);
-        dis = new LittleEndianDataInputStream(iis);
-        break;
-      default:
-        dis = new LittleEndianDataInputStream(is);
-        break;
-    }
+    if (binaryDataInfo.getCompressionType() != null) {
```

Comment:
Since you have MzMLCompressionType.NO_COMPRESSION, the binaryDataInfo.getCompressionType() should not return null. Perhaps you could declare it as @Nonnull and make sure the default value is always MzMLCompressionType.NO_COMPRESSION